File: tests/fixtures/parity/cpp/dynamic_ref.cpp
#include <iostream>
#include <map>
#include <functional>

std::map<std::string, std::function<void()>> handlers;

void register_handler(const std::string& name, std::function<void()> fn) {
    handlers[name] = fn;
}

int main() {
    register_handler("test", []() { std::cout << "Handler" << std::endl; });
    handlers["test"]();
    return 0;
}

File: tests/fixtures/parity/rust/dynamic_ref.rs
use std::collections::HashMap;

type Handler = Box<dyn Fn()>;

fn register_handler(handlers: &mut HashMap<String, Handler>, name: &str) {
    handlers.insert(name.to_string(), Box::new(|| println!("Handler")));
}

fn main() {
    let mut handlers: HashMap<String, Handler> = HashMap::new();
    register_handler(&mut handlers, "test");
}
